ChangeLog.d/m-ccm-api.txt

```diff
@@ -0,0 +1,5 @@
+Features
+   * The CCM API now supports multi-part operations. Alternative
```

Comment:
That makes sense. @mpg are you also ok to remove the change log?

Reply:
Yes, good idea. Actually I was going to comment on the ChangeLog entry as I was afraid it could confuse people, but I couldn't find a better wording to suggest, so I refrained. Omitting it entirely seems like a good option then :)

But I guess we still want a migration guide entry for the sake of alt implementors, though.

Reply:
I am not sure what to say in the migration guide. If they don't want multi-part support I think they don't have to do anything thus there is not really any migration here.

Reply:
Alt implementations must implement the whole module. We don't give them a choice.

Reply:
ok then what do they have to do ? Provide empty functions that are not called ?

Reply:
My understanding is that they have to provide working functions - unlike us, which makes it very awkward, and is why we'd like to get our own implementation as early as possible (3.1) in order to reduced the window of awkwardness.

Reply:
Though in practice until 3.1 they can probably get away with providing empty functions - but that's not future-proof, so I don't think we want to encourage that.

Reply:
Provide working functions that are not called? To reduce awkwardness we should then maybe implement the functions with just a call to the alternative implementation ?

Reply:
> 
> To reduce awkwardness we should then maybe implement the functions with just a call to the alternative implementation ?
Looking at the code I understand that this question is out of the picture.

Reply:
I'm sorry, I don't understand. We can't implement the function with just a call to the alt implementation, because (1) the alt implementation _replaces_ our implementation, and (2) most of the time there is no alt implementation.

I don't see the problem with alt implementations providing functions that are not called - we're a library, we provide functions and have no idea whether someone will call them or not. Same for the alt implementations.

Unless I misunderstood what you wrote, you might be confusing the `ALT` way with how PSA drivers work; `ALT` is much more rudimentary.

Reply:
Oh, I replied before I saw your last comment. So I guess it's all clear now.

Reply:
ok, I understand better the context now. I should be able to write a migration guide now.